Dockerfile

```diff
@@ -56,6 +56,14 @@ WORKDIR /opt/guacamole
 # Copy artifacts from builder image into this image
 COPY --from=builder /opt/guacamole/ .
 
+# Turn on the Remote IP Valve
+RUN sed -i 's+^\(      </Host>\)+        <Valve className="org.apache.catalina.valves.RemoteIpValve" />\n\1+' /usr/local/tomcat/conf/server.xml
```

Comment:
This looks like it might prove fragile, being both a regex parsing XML and a regex which is sensitive to changes in indentation. Perhaps there is a way to do this which is either aware of XML structure or aware of some degree of context? Applying a patch to `server.xml` might be an easy and more rigorous solution, if there isn't a good alternative that is XML-aware.

Reply:
I've hardened the `sed` expression to be more robust to potential indentation changes.
It's an easy one-line update, sounds now robust IMO.
Of course feel free to tell me if I should rather go with a `patch` command.
Thank you !